rtvi-client-js/src/messages.ts

```diff
@@ -159,17 +181,66 @@ export class MessageDispatcher {
     });
 
     console.debug("[MessageDispatcher] dispatch", message);
-    this._transport.sendMessage(message);
+
+    this._client.sendMessage(message);
 
     this._gc();
 
-    return promise as Promise<VoiceMessage | VoiceMessageActionResponse>;
+    return promise as Promise<RTVIMessage | RTVIMessageActionResponse>;
+  }
+
+  public async dispatchAction(
+    action: RTVIActionRequest,
+    onMessage: (message: RTVIMessage) => void
+  ): Promise<RTVIMessageActionResponse> {
+    const promise = new Promise((resolve, reject) => {
+      this._queue.push({
+        message: action,
+        timestamp: Date.now(),
+        resolve,
+        reject,
+      });
+    });
+
+    console.debug("[MessageDispatcher] action", action);
+
+    if (this._client.connected) {
```

Comment:
Race condition -- if the transport disconnects while the user is sending an action, that action will be sent to the "single turn" endpoint, rather than the intended "connected" endpoint.

Reply:
There's not really much reason to combine these into a single `action()` API, vs having `action()` and `actionSingleTurn()`, for example, which would allow the user to specify exactly what they want, without resorting to race-prone state checks.